platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -47,6 +47,46 @@ Note that certain features that rely on third-party tools, such as [PMD](https:/
 
 For details, see the full [compatibility documentation](userguide/compatibility.html#java).
 
+<a name="java-compiler-error-rendering"></a>
+### Java compiler errors in the failure report
+
+From now on, Gradle also includes Java compilation failures in the report at the end of the build.
+This will accompany the already existing errors reported when the task is executed.
+This improvement helps to identify compilation problems faster; with long executions, it can be hard to find the root cause of a compilation failure.
+
+The following is a sample output caused by a simple compilation failure:
+```
+* What went wrong:
+Execution failed for task ':project1:compileJava'.
+> Compilation failed; see the compiler output below.
+
+Java compilation error (compilation:java:java-compilation-error)
+  sample-project/src/main/java/Problem.java:6: error: incompatible types: int cannot be converted to String
+          String a = 1;
+                     ^
+```
+
+If any warning happens during the compilation, it will also be included in the report:
+```
+* What went wrong:
+Execution failed for task ':project1:compileJava'.
+> Compilation failed; see the compiler output below.
+
+Java compilation warning (compilation:java:java-compilation-warning)
+  sample-project/src/main/java/Problem1.java:6: warning: [cast] redundant cast to String
+          var warning = (String)"warning";
+                        ^
+Java compilation error (compilation:java:java-compilation-error)
+  sample-project/src/main/java/Problem2.java:6: error: incompatible types: int cannot be converted to String
+          String a = 1;
+                     ^
+```
+
+A limitation of this feature is that warnings are only reported when a failure occurs.
+We are working on additional rendering methods, which would display warnings even when the compilation is successful.
```

Comment:
This belongs to the roadmap.

```suggestion
```

Reply:
How can people find the roadmap?

Reply:
https://blog.gradle.org/roadmap-announcement

Reply:
Also, I want the users to know that we still care about reporting warnings when only warnings happen, but it is not ready yet. If you think from the user perspective, this story is not yet complete, and I want to emphasize that further features are coming.

Reply:
> https://blog.gradle.org/roadmap-announcement

Maybe I haven't asked the correct question: how the users should know that this is part of a roadmap if we don't include the roadmap.
Do yo want me to include it?

Reply:
Am I wrong not to see a roadmap item for Problems API reporting on the roadmap? I can see the Java one, but I don't see the universal reporting features.

Reply:
So my comment was about the fact that we should talk about stuff that we have in the release notes and not promise anything. Unlike the published release notes, we can adjust the roadmap later.

Reply:
Might not have enough time to make PR

Reply:
Maybe we should only mention that we'll have more enhancements as part of the [`Clean and actionable error reporting`](https://github.com/gradle/build-tool-roadmap/issues/49) roadmap item. 

Also, I'm happy to adjust the description on the roadmap based on your feedback.

Reply:
That being said, the solution uses a common rendering class: https://github.com/gradle/gradle/blob/282f62db0ac5e4f59c442229bafcb7feb0ff4593/platforms/ide/problems-rendering/src/main/java/org/gradle/problems/internal/rendering/ProblemRenderer.java#L32

I didn't make any strategy pattern here on differentiating the rendering, as I believed we agreed on an uniform look between a (coming) problem console reporting, and this report

Reply:
Pushed the new test, PTAL